File: task2/task2/src/main/java/com/task2/StockItem.java
package com.task2;

import java.io.Serializable;

public class StockItem implements Serializable{
    private static final long serialVersionUID = 1L;

    protected int quantity;
    protected Product product;

    public StockItem(){}

    public StockItem(int quantity, Product product){
        this.quantity = quantity;
        this.product = product;
    }

    // Setters
    public void setQuantity(int quantity){
        this.quantity = quantity;
    }

    public void setProduct(Product product){
        this.product = product;
    }

    // Getters
    public int getQuantity(){
        return quantity;
    }

    public Product getProduct(){
        return product;
    }

    // Other Methods
    public void updateQuantity(int newQuantity){
        quantity = newQuantity;
    }
}

File: task1/task1b/StockItem.cpp
#include "StockItem.h"
#include <iostream>

StockItem::StockItem(){}

StockItem::StockItem(int quantity, Product* product): itemQuantity(quantity), itemProduct(product){}

// Setters
void StockItem::setQuantity(int quantity){
    itemQuantity = quantity;
}

void StockItem::setProduct(Product* product){
    itemProduct = product;
}

// Getters
int StockItem::getQuantity() const{
    return itemQuantity;
}

Product* StockItem::getProduct() const{
    return itemProduct;
}

// Other Methods
void StockItem::updateQuantity(int newQuantity){
    itemQuantity += newQuantity;
}